crates/astria-sequencer-relayer/src/relayer/read.rs

```diff
@@ -233,11 +233,12 @@ async fn fetch_block(
     .await
     .wrap_err("retry attempts exhausted; bailing")?;
 
-    state.set_sequencer_connected(true);
-
-    let block = SequencerBlock::try_from_raw(block.into_inner())
-        .wrap_err("failed to parse raw proto block from grpc response")?;
+    let maybe_block = SequencerBlock::try_from_raw(block.into_inner())
+        .wrap_err("failed to parse raw proto block from grpc response");
+    state.set_sequencer_connected(maybe_block.is_ok());
 
+    let block = maybe_block?;
+    debug_assert_eq!(height, block.height());
```

Comment:
Discussed this with @joroshiba - it's maybe overkill, since it's probably reasonable to expect that responses from the sequencer don't need to be checked.  (I stumbled on this when sending a mocked response with height 6 to a request for height 1, so the `debug_assert` might help keep the mocks slightly more realistic)

Reply:
The problem is of course that we are running these tests without `debug`. But I guess we can easily change this.

However, I actually think it would be fine to add an `ensure!(height == block.height(), "requested block at height `{height}` but received a block at height `{}`", block.height());` here and just stop hard if that's violated.

conductor is doing something similar now with the responses it gets from the rollup implemeting `astria.executionapi`

Reply:
Sure - will do.

Reply:
Done in [d198996](https://github.com/astriaorg/astria/pull/1033/commits/d198996c67a81f2b6379f3c9f52d42a5ee96ba74).